lib/cassia/response_handlers/disconnect_device.rb

```diff
@@ -0,0 +1,29 @@
+module Cassia
+  module ResponseHandlers
+    class DisconnectDevice
+      def initialize(access_controller)
+        @access_controller = access_controller
+      end
+
+      def handle(response)
+        if response.success?
+          handle_success(response)
+        else
+          handle_failure(response)
+        end
+        response.success?
+      end
+
+      private
+
+      def handle_success(response)
+        that_router = @access_controller.connected_devices.find {|router| router['mac'] == response.body['device']}
```

Comment:
techincally this isn't a router it's a device, also there's a handy ruby method called `select` which you can use here to filter the list only return the devices that aren't equal to this device
```
@access_controller.connected_devices.select { |device| device.mac != response.body['device'] }
```